src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2208,35 +2265,33 @@ public object GetSocketOption(SocketOptionLevel optionLevel, SocketOptionName op
             {
                 return getIPv6MulticastOpt(optionName);
             }
-            else
-            {
-                int optionValue = 0;
 
-                // This can throw ObjectDisposedException.
-                SocketError errorCode = SocketPal.GetSockOpt(
-                    _handle,
-                    optionLevel,
-                    optionName,
-                    out optionValue);
+            int optionValue = 0;
+
+            // This can throw ObjectDisposedException.
+            SocketError errorCode = SocketPal.GetSockOpt(
+                _handle,
+                optionLevel,
+                optionName,
+                out optionValue);
 
-                GlobalLog.Print("Socket#" + Logging.HashString(this) + "::GetSocketOption() Interop.Winsock.getsockopt returns errorCode:" + errorCode);
+            GlobalLog.Print("Socket#" + Logging.HashString(this) + "::GetSocketOption() Interop.Winsock.getsockopt returns errorCode:" + errorCode);
 
+            //
+            // if the native call fails we'll throw a SocketException
+            //
+            if (errorCode != SocketError.Success)
+            {
                 //
-                // if the native call fails we'll throw a SocketException
+                // update our internal state after this socket error and throw
```

Comment:
Nit: capitalize comment and end in "." (Although I presume one of us will have to go through the rest of the code and do that for cleanup.)